security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -211,6 +218,14 @@ func (sc *SecretManagerClient) OnSecretUpdate(resourceName string) {
 	}
 }
 
+// Reset clears the cache and cancels any pending certificate rotation events.
+func (sc *SecretManagerClient) Reset() {
```

Comment:
Is it beneficial to clear the cache vs invalidate it?

Reply:
Otherwise you need to rollout/restart all wordloads?